components/entry/entry-form.tsx

```diff
@@ -167,10 +181,36 @@ const ListField = ({
               <SortableContext items={arrayFields.map(item => item.id)} strategy={verticalListSortingStrategy}>
                 {arrayFields.map((arrayField, index) => (
                   <SortableItem key={arrayField.id} id={arrayField.id} type={field.type}>
-                    <div className="grid gap-6 flex-1">
-                      {field.type === "object" && field.fields
-                        ? renderFields(field.fields, `${fieldName}.${index}`)
-                        : renderSingleField(field, `${fieldName}.${index}`, control, false)}
+                    {collapsible && (
+                      <Tooltip>
+                        <TooltipTrigger asChild>
+                          <Button type="button" variant="ghost" size="icon-sm" className="h-8" onClick={() => toggleCollapse(arrayField.id)}>
+                            {expandedFields.includes(arrayField.id) ? (
+                              <ChevronsDownUp className="h-4 w-4" />
+                            ) : (
+                              <ChevronsUpDown className="h-4 w-4" />
+                            )}
+                          </Button>
+                        </TooltipTrigger>
+                        <TooltipContent>
+                          {expandedFields.includes(arrayField.id) ? 'Collapse' : 'Expand'}
+                        </TooltipContent>
+                      </Tooltip>
+                    )}
+                    <div className="grid gap-3 flex-1">
+                      {collapsible && (
+                          <div className="mt-1 pl-1">
+                            <p className="text-base font-medium text-gray-900 dark:text-gray-100">
+                              {/* TODO: Make stateful - update as user types, and after save (without reload) */}
+                              {typeof collapsible === 'object' ? generateFieldLabel(collapsible.label ?? `Item ${index + 1}`, arrayField) : `Item ${index + 1}`}
```

Comment:
Unsure about defualt "Item #", because number changes during re-sort, which doesnt feel right.


https://github.com/user-attachments/assets/91579192-ad10-41cd-8b75-f8b8055bc489

Reply:
Couldnt think of better approach, so I did this:

![CleanShot 2025-04-18 at 11 09 59@2x](https://github.com/user-attachments/assets/3c7fd1b5-3a0f-4f5a-8557-f050c82d7c09)